### PR TITLE
Update FASTP to include adapter list file

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by matthdsm — APPROVED** [review]

File: modules/nf-core/fastp/main.nf [hunk 1]
```diff
@@ -27,6 +28,7 @@ process FASTP {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def adapterlist = adapter_fasta ? "--adapter_fasta ${adapter_fasta}" : ""
```
Comment (JoseEspinosa) [inline modules/nf-core/fastp/main.nf:31]:
```suggestion
    def adapter_list = adapter_fasta ? "--adapter_fasta ${adapter_fasta}" : ""
```

File: modules/nf-core/fastp/main.nf [hunk 2]
```diff
@@ -40,6 +42,7 @@ process FASTP {
             --thread $task.cpus \\
             --json ${prefix}.fastp.json \\
             --html ${prefix}.fastp.html \\
+            $adapterlist \\
```
Comment (JoseEspinosa) [inline modules/nf-core/fastp/main.nf:45]:
```suggestion
          $adapter_list \\
```

File: modules/nf-core/fastp/main.nf [hunk 3]
```diff
@@ -61,6 +64,7 @@ process FASTP {
...
```
Comment (JoseEspinosa) [inline modules/nf-core/fastp/main.nf:67]:
```suggestion
            $adapter_list \\
```

File: modules/nf-core/fastp/main.nf [hunk 4]
```diff
@@ -82,6 +86,7 @@ process FASTP {
             --out2 ${prefix}_2.fastp.fastq.gz \\
             --json ${prefix}.fastp.json \\
             --html ${prefix}.fastp.html \\
+            $adapterlist \\
```
Comment (JoseEspinosa) [inline modules/nf-core/fastp/main.nf:89]:
```suggestion
            $adapter_list \\
```

**Review by JoseEspinosa — APPROVED** [review]
LGTM! 🚀 Just a small suggestion feel free to accept it or not 😁

File: tests/modules/nf-core/fastp/main.nf [hunk 5]
```diff
@@ -38,10 +40,11 @@ workflow test_fastp_interleaved {
     input = [ [ id:'test', single_end:true ], // meta map
               [ file(params.test_data['sarscov2']['illumina']['test_interleaved_fastq_gz'], checkIfExists: true) ]
             ]
+    adapter_fasta = []
     save_trimmed_fail = false
     save_merged       = false
 
-    FASTP ( input, save_trimmed_fail, save_merged )
+    FASTP ( input, adapter_fasta, save_trimmed_fail, save_merged )
```
Comment (Emiller88) [inline tests/modules/nf-core/fastp/main.nf:47]:
```suggestion
    FASTP ( input, [], save_trimmed_fail, save_merged )
```
Can we test it like this so the empty list `[]` is really obvious?

**Review by Emiller88 — APPROVED** [review]

**Pull request merged**